python/ray/tests/test_gcs_fault_tolerance.py

```diff
@@ -214,6 +215,72 @@ def test_gcs_client_reconnect(ray_start_regular_with_external_redis, auto_reconn
         assert gcs_client.internal_kv_get(b"a", None) == b"b"
 
 
+@pytest.mark.parametrize(
+    "ray_start_regular_with_external_redis",
+    [
+        {
+            **generate_system_config_map(
+                num_heartbeats_timeout=20, gcs_rpc_server_reconnect_timeout_s=3600
+            ),
+            "namespace": "actor",
+        }
+    ],
+    indirect=True,
+)
+def test_actor_workloads(ray_start_regular_with_external_redis):
+    """This test cover the case to create actor while gcs is down
+    and also make sure existing actor continue to work even when
+    GCS is down.
+    """
+
+    @ray.remote
+    class Counter:
+        def r(self, v):
+            return v
+
+    c = Counter.remote()
+    r = ray.get(c.r.remote(10))
+    print("c.r.remote(10)", r)
+    assert r == 10
+
+    print("GCS is killed")
+    ray.worker._global_node.kill_gcs_server()
+
+    print("Start to create a new actor")
+    cc = Counter.remote()
+    with pytest.raises(ray.exceptions.GetTimeoutError):
+        ray.get(cc.r.remote(10), timeout=5)
+
+    ray.worker._global_node.start_gcs_server()
+
+    r = ray.get(cc.r.remote(10))
+    assert r == 10
+    print("cc.r.remote(10)", r)
+
+    c = Counter.options(lifetime="detached", name="C").remote()
+
+    assert ray.get(c.r.remote(10)) == 10
+
+    ray.worker._global_node.kill_gcs_server()
+
+    sleep(2)
+
+    assert ray.get(c.r.remote(10)) == 10
```

Comment:
> Can you add a test if GCS fails and "not recovered" (which is the common path for non HA), APIs will fail properly and things won't be hanging?


@rkooo567 it's been covered here.

Reply:
To be honest, it's not covering everything e2e because we also have other issues.

The main goal of this PR is to make sure the channel is reconnected and cpp test should cover this one.

The e2e ci test here is to cover the case where function is broken because of reconnection is not working properly.